Add metabomics-lcms-preprocessing WF (#292)
galaxyproject/iwc: workflows/metabomics-lcms-preprocessing/README.md

# Mass spectrometry: LC-MS preprocessing with XCMS 

> This workflow is composed with the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) able to extract, filter, align and fill gapand the possibility to annotate isotopes, adducts and fragments using the CAMERA R package [(Kuhl, C 2012)](https://bioconductor.org/packages/release/bioc/html/CAMERA.html).

Reviewer: ```suggestion
This workflow uses the XCMS tool R package [(Smith, C.A. 2006)](https://bioconductor.org/packages/release/bioc/html/xcms.html) to extract, filter, align and fill gaps, and uses the CAMERA R package [(Kuhl, C 2012)](https://bioconductor.org/packages/release/bioc/html/CAMERA.html) to annotate isotopes, adducts and fragments .
```

> 🎓 For more information, it is explained in the [Galaxy Training Network tutorial: Mass spectrometry: LC-MS preprocessing with XCMS](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html)

Reviewer: ```suggestion
🎓 For more information see the [Galaxy Training Network tutorial: Mass spectrometry: LC-MS preprocessing with XCMS](https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html)
```

## Inputs
### sampleMetadata
The sampleMetadata tabular file corresponds to a table containing information about your samples

A sample metadata file contains various information for each of your raw files:
- Classes which will be used during the preprocessing steps
- Analytical batches which will be useful for a batch correction step, along with sample types (pool/sample) and injection order
- Different experimental conditions which can be used for statistics
- Any information about samples that you want to keep, in a column format

The content of your sample metadata file has to be filled by you, since it is not contained in your raw data. Note that you can either:
- Upload an existing metadata file

> - Use a template to create one (because it can be painful to get the sample list without misspelling or omission)

Reviewer: Does it make sense to provide a link to a template here (or link to the corresponding tutorial section) ?

Author: As explained the line after, the template can be generated with the xcms get a sampleMetadata file tool.

The tool will fill the samplename based on the infile names.

Author: > Would it make sense to create a metabolomics folder under workflows/ ?
Done 👍

  - Generate a template with the xcms get a sampleMetadata file tool tool
  - Fill it using your favorite table editor (Excel, LibreOffice)
  - Upload it within Galaxy

**Formats:** tab-separated values as tsv, tab, txt, ...

### Chromatograms Dataset Collection
Mass-spectrometry data files gathered in a Galaxy Dataser Collection

**Formats:** open format as mzXML, mzMl, mzData and netCDF

## Main steps
1. MSnbase readMSData: read the mzXML and prepare for xcms
2. XCMS findChromPeaks: peak picking
3. XCMS groupChromPeaks: determining shared ions across samples
4. XCMS adjustRtime: retention time correction
5. XCMS fillChromPeaks: integrating areas of missing peaks
6. CAMERA.annotate: annotation

> 

Reviewer: ```suggestion
```
